pylint/checkers/exceptions.py

```diff
@@ -359,6 +365,29 @@ def visit_tryexcept(self, node):
 
                     exc_ancestors = [anc for anc in exc.ancestors()
                                      if isinstance(anc, astroid.ClassDef)]
+
+                    for prev_part, prev_exc in handled_in_clause:
+                        if exc == prev_exc:
+                            self.add_message('overlapping-except',
```

Comment:
Can we do the checks first and then, if one of the checks matches, can we emit the message? This will eliminate some redundancy.